src/components/utils/include/utils/timer.h

```diff
@@ -197,19 +198,19 @@ class Timer {
   };
 
   const std::string name_;
-  const TimerTask* task_;
+  std::auto_ptr<TimerTask> task_;
   uint32_t timeout_ms_;
   sync_primitives::atomic_bool is_running_;
-  threads::Thread* thread_;
-  TimerDelegate* delegate_;
+  TimerDelegate delegate_;
+  threads::Thread *thread_;
```

Comment:
Because it use proprietary methods for allocate and delete memory. `DeleteThread() and threads::CreateThread(name_.c_str(), &delegate_).`